### PR TITLE
[IMP] *_*: transfer the custom information

### PR DESCRIPTION
**Pull request by mitali-willdooit**
Description of the issue/feature this PR addresses:
<b>Task: </b>https://www.odoo.com/web?#id=1970476&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
<b>Pad: </b> https://pad.odoo.com/p/r.c7185b654635d21d65faae0cd5facc04
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -238,6 +238,7 @@ def _compute_confirm_cancel(self):
                 productions_with_done_move[production_record[0]] = True
         for production in self:
             production.confirm_cancel = productions_with_done_move.get(production.id, False)
+    description = fields.Char()
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_production.py]: can you move this definition with the other fields ?
it's also missing an help

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -965,6 +966,7 @@ def _prepare_procurement_values(self):
             elif self.rule_id.group_propagation_option == 'none':
                 group_id = False
         return {
+            'description': self.description or self.name.replace(self.product_id.display_name, ''),
```
Comment (sle-odoo) [inline addons/stock/models/stock_move.py]: why not use description_picking here?

File: addons/stock/models/stock_rule.py [hunk 3]
```diff
@@ -282,7 +284,7 @@ def _get_stock_move_values(self, product_id, product_qty, product_uom, location_
             'propagate_cancel': self.propagate_cancel,
             'propagate_date': self.propagate_date,
             'propagate_date_minimum_delta': self.propagate_date_minimum_delta,
-            'description_picking': product_id._get_description(self.picking_type_id),
+            'description_picking': values.get('description') and (product_desc + values.get('description')) or product_desc,
```
Comment (sle-odoo) [inline addons/stock/models/stock_rule.py]: seems very weird, you added the description field but you changed the use of description_picking?

File: addons/purchase_stock/tests/test_purchase_lead_time.py [hunk 4]
```diff
@@ -239,3 +239,72 @@ def test_merge_po_line_2(self):
         self.assertEqual(len(po_line), 2, 'the purchase order lines are not merged')
         self.assertEqual(po_line[0].product_qty, 10, 'the purchase order line has a wrong quantity')
         self.assertEqual(po_line[1].product_qty, 5, 'the purchase order line has a wrong quantity')
+
+    def test_merge_po_line_3(self):
+        """Chage that merging po line for same procurement is done depending on
+        propagate_date and description"""
+
+        # create a product with manufacture route
```
Comment (sle-odoo) [inline addons/purchase_stock/tests/test_purchase_lead_time.py]: what are you testing? you don't even use the description field you added on the move?

File: addons/stock/models/stock_rule.py (hunk 3 above)
Comment (hpr-odoo) [reply addons/stock/models/stock_rule.py]: done

File: addons/purchase_stock/tests/test_purchase_lead_time.py (hunk 4 above)
Comment (hpr-odoo) [reply addons/purchase_stock/tests/test_purchase_lead_time.py]: done

File: addons/mrp/models/mrp_production.py (hunk 1 above)
Comment (hpr-odoo) [reply addons/mrp/models/mrp_production.py]: done

File: addons/stock/models/stock_move.py (hunk 2 above)
Comment (hpr-odoo) [reply addons/stock/models/stock_move.py]: when duplicate the SO(which has custom detail in SOline) and confirm it then description picking will have only product name without custom detail.

File: addons/purchase_stock/models/purchase.py [hunk 5]
```diff
@@ -404,5 +404,5 @@ def _find_candidate(self, product_id, product_qty, product_uom, location_id, nam
         args can be merged. If it returns an empty record then a new line will
         be created.
         """
-        lines = self.filtered(lambda l: l.propagate_date == values['propagate_date'] and l.propagate_date_minimum_delta == values['propagate_date_minimum_delta'] and l.propagate_cancel == values['propagate_cancel'])
+        lines = self.filtered(lambda l: l.propagate_date == values['propagate_date'] and l.propagate_date_minimum_delta == values['propagate_date_minimum_delta'] and l.propagate_cancel == values['propagate_cancel'] and l.name == (l.product_id.display_name+values['description']))
```
Comment (sle-odoo) [inline addons/purchase_stock/models/purchase.py]: weird

File: addons/purchase_stock/models/purchase.py [hunk 6]
```diff
@@ -239,6 +240,7 @@ class PurchaseOrderLine(models.Model):
     propagate_date = fields.Boolean(string="Propagate Rescheduling", help='The rescheduling is propagated to the next move.')
     propagate_date_minimum_delta = fields.Integer(string='Reschedule if Higher Than', help='The change must be higher than this value to be propagated')
     propagate_cancel = fields.Boolean('Propagate cancellation', default=True)
+    product_custom_attribute_values = fields.Text()
```
Comment (amoyaux) [inline addons/purchase_stock/models/purchase.py]: remove

File: addons/purchase_stock/models/purchase.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+import json
```
Comment (amoyaux) [inline addons/purchase_stock/models/purchase.py]: remove

File: addons/purchase_stock/models/purchase.py [hunk 8]
```diff
@@ -267,6 +269,12 @@ def _compute_qty_received(self):
                             total += move.product_uom._compute_quantity(move.product_uom_qty, line.product_uom)
                 line.qty_received = total
 
+    @api.onchange('product_id')
```
Comment (amoyaux) [inline addons/purchase_stock/models/purchase.py]: remove

File: addons/purchase_stock/models/purchase.py [hunk 9]
```diff
@@ -397,6 +405,8 @@ def _prepare_purchase_order_line_from_procurement(self, product_id, product_qty,
         res['propagate_cancel'] = values.get('propagate_cancel')
         res['propagate_date'] = values.get('propagate_date')
         res['propagate_date_minimum_delta'] = values.get('propagate_date_minimum_delta')
+        res['product_custom_attribute_values'] = values.get('product_custom_attribute_values')
```
Comment (amoyaux) [inline addons/purchase_stock/models/purchase.py]: remove

File: addons/purchase_stock/models/purchase.py [hunk 10]
```diff
@@ -397,6 +405,8 @@ def _prepare_purchase_order_line_from_procurement(self, product_id, product_qty,
...
+        res['name'] = product_id.get_product_multiline_description_sale() + values.get('description', '')
```
Comment (amoyaux) [inline addons/purchase_stock/models/purchase.py]: upper:
```
name = product_id.get_product_multiline_description_sale()
if values.get('product_custom_attribute_values'):
    name += '\n' + values['product_custom_attribute_values']
```

here
```
res['name'] = name
```

File: addons/mrp/models/mrp_production.py [hunk 11]
```diff
@@ -220,6 +220,7 @@ def _get_default_date_planned_finished(self):
     picking_ids = fields.Many2many('stock.picking', compute='_compute_picking_ids', string='Picking associated to this manufacturing order')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
     confirm_cancel = fields.Boolean(compute='_compute_confirm_cancel')
+    description = fields.Char('Custom Description')
```
Comment (amoyaux) [inline addons/mrp/models/mrp_production.py]: finished_product_custom_attribute_values

File: addons/mrp/models/stock_rule.py [hunk 12]
```diff
@@ -77,6 +77,7 @@ def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, na
         return {
             'origin': origin,
             'product_id': product_id.id,
+            'description': values.get('description'),
```
Comment (amoyaux) [inline addons/mrp/models/stock_rule.py]: 'finished_product_custom_attribute_values':  values.get('product_custom_attribute_values'),

File: addons/sale/models/sale.py [hunk 13]
```diff
@@ -1708,7 +1708,7 @@ def _get_sale_order_line_multiline_description_variants(self):
 
         # display the no_variant attributes, except those that are also
         # displayed by a custom (avoid duplicate)
-        for ptav in (self.product_no_variant_attribute_value_ids - custom_values):
+        for ptav in self.product_no_variant_attribute_value_ids.filtered(lambda x: x.ids[0] not in custom_values.ids):
```
Comment (amoyaux) [inline addons/sale/models/sale.py]: remove this commit fixed in https://github.com/odoo/odoo/pull/42201

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
rebase-ff

**Comment by Feyensv** [comment]
@robodoo rebase-ff
;)

**Comment by amoyaux** [comment]
@Feyensv oups, merci

**Pull request closed without merging**